### PR TITLE
feat: introduce proxy module

### PR DESCRIPTION
**Pull request by chunshao90**
## Which issue does this PR close?

Closes #

## Rationale for this change
 Introduce proxy module. Refer to #681 .

## What changes are included in this PR?
* Introduce proxy module.
* Impl storage_service with proxy.
* Support forwarding stream query.

## Are there any user-facing changes?
No.

## How does this change test
Integration test.

### CODE REVIEW

File: docs/minimal.toml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+[node]
+addr = "ceresdb0"
```
Comment (ShiKaiWi) [inline docs/minimal.toml:2]: It is weird to use a name for addr.

File: server/src/proxy/mod.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+pub(crate) mod error;
```
Comment (jiacai2050) [inline server/src/proxy/mod.rs:3]: Add crates description explain what it does

File: docs/minimal.toml (hunk 1 above)
Comment (jiacai2050) [reply docs/minimal.toml:2]: is this a required config, even for standalone mode?

File: server/src/proxy/forward.rs [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+    /// default keep http2 connections alive while idle
+    pub keep_alive_while_idle: bool,
+    pub connect_timeout: Duration,
+    pub forward_timeout: Duration,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            enable: false,
```
Comment (ShiKaiWi) [inline server/src/proxy/forward.rs:103]: Shall we deprecate this option? It should be always enabled.

File: server/src/proxy/error.rs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+use crate::error_util;
+
+define_result!(Error);
+
+#[derive(Snafu, Debug)]
+#[snafu(visibility(pub))]
+pub enum Error {
+    #[snafu(display("Internal error, message:{}, cause:{}", msg, source))]
+    Internal { msg: String, source: GenericError },
```
Comment (ShiKaiWi) [inline server/src/proxy/error.rs:16]: Any differences between `Internal` and `ErrNoCause`?

File: server/src/proxy/grpc/sql_query.rs [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+        let req = match self.maybe_forward_sql_query(&req).await {
+            Some(resp) => return resp,
+            None => req,
+        };
+
+        let output = self.fetch_sql_query_output(ctx, &req).await?;
+        convert_output(&output, self.resp_compress_min_length)
+    }
+
+    async fn handle_stream_query_internal(
```
Comment (ShiKaiWi) [inline server/src/proxy/grpc/sql_query.rs:82]:
```suggestion
    async fn handle_stream_sql_query_internal(
```

**Review by ShiKaiWi — APPROVED** [review]
LGTM

**Pull request merged**